crypto/asn1/a_sign.c

```diff
@@ -198,10 +203,16 @@ int ASN1_item_sign_ctx(const ASN1_ITEM *it,
 
     }
 
-    inl = ASN1_item_i2d(asn, &buf_in, it);
+    buf_len = ASN1_item_i2d(asn, &buf_in, it);
+    if (buf_len <= 0) {
```

Comment:
Similarly here. If `buf_len` < 0 then `OPENSSL_clear_free` does the wrong thing.

Reply:
Fixed below in code, by calling free  with an separated variable:
`   OPENSSL_clear_free((char *)buf_in, inl);`